libs/platform/auth/src/oauth/oauth.service.ts

```diff
@@ -192,11 +203,20 @@ export class OauthService implements AuthService, AuthTokenService {
     return this.state$;
   }
 
-  protected restoreState(): void {
-    this.storageService
+  protected initState(): Observable<OauthServiceState | null> {
+    return this.storageService
       .get<OauthServiceState>(OauthService.STATE_KEY)
-      .pipe(distinctUntilChanged())
-      .subscribe((state) => this.state$.next({ ...state }));
+      .pipe(
+        distinctUntilChanged(),
+        map((state) => {
+          this.state$.next({ ...state });
+          return state;
+        })
+      );
+  }
+
+  protected restoreState(): void {
```

Comment:
Do we still need it and why?

Reply:
It's still being used by `invokeStoredToken`, but I can move the implementation of `restoreState` there so we won't need it anymore.

Reply:
yes, let's do this small refactor and move implementation to `invokeStoredToken`